pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -21,10 +25,59 @@ type DualReadWriter struct {
 	repo    repository.ReaderWriter
 	parser  Parser
 	folders *FolderManager
+	access  authlib.AccessChecker
+}
+
+func NewDualReadWriter(repo repository.ReaderWriter, parser Parser, folders *FolderManager, access authlib.AccessChecker) *DualReadWriter {
+	return &DualReadWriter{repo: repo, parser: parser, folders: folders, access: access}
+}
+
+func (r *DualReadWriter) authorize(ctx context.Context, parsed *ParsedResource, verb string) error {
```

Comment:
maybe at the bottom of the file as they are private and less relevant than the other operations?